src/bit-fields/src/example.rs

```diff
@@ -0,0 +1,908 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate self as bit_fields;
+/// Generated by:
+/// ```ignore
+/// bit_fields::bitfield!(ExampleBitField,u32,{
+///     /// RANGE1 bit field
+///     RANGE1: 0..1,
+///     /// SSE bit flag
+///     SSE: 2,
+///     /// SSE1 bit flag
+///     SSE1: 3,
+///     /// RANGE2 bit field
+///     RANGE2: 4..6,
+///     /// SSE2 bit flag
+///     SSE2: 9,
+///     /// SSE3 bit flag
+///     SSE3: 10,
+///     /// RANGE3 bit field
+///     RANGE3: 12..15,
+///     /// SSE4 bit flag
+///     SSE4: 18,
+/// });
+/// ```
+/// Everything below this line was generated.
+///
+/// ---
+///
+/// A 32 bit structure containing a number of bit flags and bit fields.
+///
+/// ## Layout
+///
+/// <table>
+///     <tr><th>Bit/s</th><th>Identifier</th><th>Description</th></tr>
+///     <tr><td>00..01</td><td>RANGE1</td><td>RANGE1 bit field </td></tr>
+///     <tr><td>02</td><td>SSE</td><td>SSE bit flag </td></tr>
+///     <tr><td>03</td><td>SSE1</td><td>SSE1 bit flag </td></tr>
+///     <tr><td>04..06</td><td>RANGE2</td><td>RANGE2 bit field </td></tr>
+///     <tr><td>09</td><td>SSE2</td><td>SSE2 bit flag </td></tr>
+///     <tr><td>10</td><td>SSE3</td><td>SSE3 bit flag </td></tr>
+///     <tr><td>12..15</td><td>RANGE3</td><td>RANGE3 bit field </td></tr>
+///     <tr><td>18</td><td>SSE4</td><td>SSE4 bit flag </td></tr>
+/// </table>
+#[derive(Clone)]
+#[repr(C)]
+pub struct ExampleBitField {
```

Comment:
I am not aware of this file being used or required for the Firecracker process' runtime operations. If that is true, can this be moved either into the FC demo project or into a unit test file and made use of in that context instead? (See my comment on `main.rs`)

Reply:
https://github.com/firecracker-microvm/firecracker/pull/3105#discussion_r969304852